plugins/org/src/components/MyGroupsSidebarItem/MyGroupsSidebarItem.test.tsx

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { renderInTestApp, TestApiProvider } from '@backstage/test-utils';
+import React from 'react';
+import { MyGroupsSidebarItem } from './MyGroupsSidebarItem';
+import GroupIcon from '@material-ui/icons/People';
+import { IdentityApi, identityApiRef } from '@backstage/core-plugin-api';
+import { CatalogApi } from '@backstage/catalog-client';
+import { Entity } from '@backstage/catalog-model';
+import { catalogApiRef, entityRouteRef } from '@backstage/plugin-catalog-react';
+
+describe('MyGroupsSidebarItem Test', () => {
+  describe('For guests or users with no groups', () => {
+    it('MyGroupsSidebarItem should be empty', async () => {
+      const identityApi: Partial<IdentityApi> = {
+        getBackstageIdentity: async () => ({
+          type: 'user',
+          userEntityRef: 'user:default/guest',
+          ownershipEntityRefs: ['user:default/guest'],
+        }),
+      };
+      const catalogApi: Partial<CatalogApi> = {
+        getEntities: () =>
+          Promise.resolve({
+            items: [] as Entity[],
+          }),
+      };
+      const rendered = await renderInTestApp(
+        <TestApiProvider
+          apis={[
+            [identityApiRef, identityApi],
+            [catalogApiRef, catalogApi],
+          ]}
+        >
+          <MyGroupsSidebarItem
+            props={{
+              singularTitle: 'My Squad',
+              pluralTitle: 'My Squads',
+              icon: GroupIcon,
+            }}
+          />
+        </TestApiProvider>,
+        {
+          mountedRoutes: {
+            '/': entityRouteRef,
+          },
```

Comment:
I needed to add this in as I'm using `useRouteRef(entityRouteRef)` to get the proper route but it always seems to output something into the results in this case "Mounted at /". How do I get it to stop doing that? Checking for this string seems wrong as that's now how it would really work. @freben any suggestions?

Reply:
I could swear that using `/` caused my test to pass locally though, but they are currently failing :(

Reply:
I might be missing some context that @freben has here but there are existing tests using entityRouteRef that mount it like this. 

```ts
        mountedRoutes: {
          '/catalog/:namespace/:kind/:name': entityRouteRef,
        },
```

Reply:
Yes, I'm using that in 2 of the 3 tests for this feature. The other 2 tests I have elements I can check to validate the expected results. In this case of this tests it is looking at the case where the user is not logged in or guest or has no groups, the expected result would be and empty. I did initially put exactly what you mentioned above but then I get "Mounted at /catalog/:namespace/:kind/:name" in my results which is not an empty element 😢

Reply:
Yeah I checked out and looked at this. It's just the `.toBeEmptyDOMElement` expectation that's wrong. If you slap a `rendered.debug();` after rendering in any of the other tests, you'll see that the output is like

```tsx
      <body>
        <div>
          <div>
            Mounted at
            /catalog/:namespace/:kind/:name
          </div>
          <a
            aria-label="My Squad"
            class="BackstageSidebarItem-root-1 BackstageSidebarItem-closed-3 BackstageSidebarItem-highlightable-5"
            href="/catalog/default/Group/team-a"
          >
            <span
              class="MuiBadge-root BackstageSidebarItem-closedItemIcon-14"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-anchorOriginTopRightCircular MuiBadge-colorSecondary MuiBadge-invisible MuiBadge-dot"
              />
            </span>
          </a>
        </div>
      </body>
```

Note the "Mounted at" at the top. So your assertion should probably be something like "there's only one div and its textual content is `Mounted at /catalog/:namespace/:kind/:name`" or something like that, if anything

Reply:
Thanks, did not know about `rendered.deubg()`, that's pretty handy. I've added a check for the text, could not figure out a way to check for a single `div` though :(